src/modm/platform/dma/stm32/module.lb

```diff
@@ -47,17 +79,42 @@ def build(env):
         max_channels = 0
         for channel in channels["channel"]:
             max_channels = channel["position"]
-            for request in channel["request"]:
-                for signal in request["signal"]:
+            if dma["type"] in ["stm32-channel-request"]:
+                for request in channel["request"]:
+                    for signal in request["signal"]:
+                        if "name" in signal:
+                            signal_name = signal["name"].capitalize()
+                            signal_names[signal_name] = 1
+            else:
+                for signal in channel["signal"]:
                     if "name" in signal:
                         signal_name = signal["name"].capitalize()
                         signal_names[signal_name] = 1
         controller.append({"instance": int(channels["instance"]), "channels": int(max_channels)})
 
+    did = device.identifier
+    if (did.family in ['f0'] and did.name == '30' and did.size == 'c') or (did.family in ['f1'] and did.name == '02'):
+        print("FIXME: Bug in modm-deviced data: Dma2 does not exist on " + str(device.identifier))
```

Comment:
Use `env.log.error()` or `env.log.warning()` so you know what module this is coming from.

Reply:
Converted to python comment as it is actually only relevant for developers.